File: SlSdkDemo/SdkContract/src/main/java/com/bmtc/sdk/contract/base/BaseFragment.java
package com.bmtc.sdk.contract.base;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;



/**
 * Created by zj on 2017/10/18.
 */

public class BaseFragment extends Fragment{

    private boolean mHidden = false;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onResume() {
        super.onResume();
       // MobclickAgent.onPageStart(this.getClass().getSimpleName());
    }

    @Override
    public void onPause() {
        super.onPause();
       // MobclickAgent.onPageEnd(this.getClass().getSimpleName());
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);
        mHidden = !isVisibleToUser;
    }

    public boolean getHidden() {
        BaseFragment fragment = (BaseFragment) getParentFragment();
        if (fragment != null && fragment.getHidden()) {
            return true;
        }

        return mHidden;
    }

}

File: SlSdkDemo/app/src/main/java/com/yjkj/chainup/base/BaseFragment.kt
package com.yjkj.chainup.base

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * @Author: Bertking
 * @Date：2019-06-15-10:48
 * @Description:
 */
abstract class BaseFragment : Fragment() {
    val TAG = this::class.java.simpleName

    var rootView: View? = null
    abstract fun setLayoutId(): Int

    abstract fun initData()

    abstract fun initView()


    override fun onAttach(context: Context?) {
        super.onAttach(context)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initData()
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.d(TAG,"======onViewCreated=======")
        initView()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater.inflate(setLayoutId(), container, false)
        Log.d(TAG, "======rootView:${rootView == null}====")
        return rootView
    }


    override fun onDestroy() {
        super.onDestroy()
    }

}